app/services/migration/migrators/declaration.rb

```diff
@@ -37,6 +37,7 @@ def call
           cohort_id: find_cohort_id!(ecf_id: ecf_declaration.cohort_id),
           lead_provider_id: find_lead_provider_id!(ecf_id: ecf_declaration.cpd_lead_provider.npq_lead_provider.id),
           application_id: find_application_id!(course_identifier: ecf_declaration.course_identifier, ecf_user_id: ecf_declaration.user_id),
+          skip_declaration_date_within_schedule_validation:,
```

Comment:
Instead of adding `attr_accessor` on `Declaration` model, as this is a "special case" I feel it might be better to only make change here on the migration by doing:
```ruby
        declaration.attributes = {
          created_at: ecf_declaration.created_at,
          updated_at: ecf_declaration.updated_at,
          declaration_type: ecf_declaration.declaration_type,
          declaration_date: ecf_declaration.declaration_date,
          state: ecf_declaration.state,
          state_reason: latest_declaration_state&.state_reason,
          cohort_id: find_cohort_id!(ecf_id: ecf_declaration.cohort_id),
          lead_provider_id: find_lead_provider_id!(ecf_id: ecf_declaration.cpd_lead_provider.npq_lead_provider.id),
          application_id: find_application_id!(course_identifier: ecf_declaration.course_identifier, ecf_user_id: ecf_declaration.user_id),
        }
        declaration.save!(validate: false)
```

What do you think?

Reply:
passing `validate: false` on the save is dangerous as it will skip _all_ validation; we still want failures if the declaration is invalid according to any other rules

Reply:
agree around wanting to keep the other validations and just ignoring this one